chrome/browser/net/chrome_network_delegate.cc

```diff
@@ -327,6 +330,95 @@ int ChromeNetworkDelegate::OnBeforeURLRequest(
   return rv;
 }
 
+std::string generateBlockedElemhideJavaScript(std::string url) {
+  std::stringstream ss;
+
+  // basically copy-pasting JS from
+  // https://github.com/adblockplus/adblockpluschrome/blob/master/include.preload.js#L546
+  ss <<
+"function hideElement(element)\n\
+{\n\
+  function doHide()\n\
+  {\n\
+    let propertyName = \"display\";\n\
+    let propertyValue = \"none\";\n\
+    if (element.localName == \"frame\")\n\
+    {\n\
+      propertyName = \"visibility\";\n\
+      propertyValue = \"hidden\";\n\
+    }\n\
+\n\
+    if (element.style.getPropertyValue(propertyName) != propertyValue ||\n\
+        element.style.getPropertyPriority(propertyName) != \"important\")\n\
+      element.style.setProperty(propertyName, propertyValue, \"important\");\n\
+  }\n\
+\n\
+  doHide();\n\
+\n\
+  new MutationObserver(doHide).observe(\n\
+    element, {\n\
+      attributes: true,\n\
+      attributeFilter: [\"style\"]\n\
+    }\n\
+  );\n\
+}";
+
+  ss <<
+"\n\
+ var selector = \"[src='" + url + "'], [srcset='" + url + "']\";\n\
+ //console.log('blocked element selector: ' + selector);\n\
+ var elements = document.querySelectorAll(selector);\n\
+ if (elements.length > 0) {\n\
+   //console.log('found ' + elements.length + ' elemhide blocked elements for selector ' + selector);\n\
+   for (var e=0; e<elements.length; e++) {\n\
+     hideElement(elements[e]);\n\
```

Comment:
i can only `hideElement` and in original sources it's not used sometimes:
https://github.com/adblockplus/adblockpluschrome/blob/master/include.preload.js#L224